action.yaml

```diff
@@ -49,35 +54,52 @@ runs:
         fetch-depth: ${{ inputs.REPO_DEPTH }}
         ref: ${{ inputs.VERSION_TAG }}
         repository: ${{ inputs.REPOSITORY }}
+    
     - name: Git Config
       run: |
         git config --global user.name "GitHub Actions"
         git config --global user.email "actions@users.noreply.github.com"
       shell: bash
+
+    - name: Install pnpm
+      if: inputs.NODE_PACKAGE_MANAGER == 'pnpm'
+      uses: pnpm/action-setup@v3
+      with:
+        version: 8
+    
     - name: Setup Nodejs
       uses: actions/setup-node@v4
       if: inputs.DISABLE_NPM_CACHE == 'false'
       with:
         node-version: ${{ inputs.NODE_VERSION }}
-        cache: 'npm'
-        cache-dependency-path: '**/package-lock.json'
+        cache: ${{ inputs.NODE_PACKAGE_MANAGER }}
+        # Used to specify the path to a dependency file: package-lock.json, yarn.lock, etc. 
+        # It will generate hash from the target file for primary key. It works only If cache is specified.  
+        # Supports wildcards or a list of file names for caching multiple dependencies.
+        # Default: ''
+        # cache-dependency-path: '**/package-lock.json'
     - name: Setup Nodejs Without Cache
       uses: actions/setup-node@v4
       if: inputs.DISABLE_NPM_CACHE == 'true'
       with:
         node-version: ${{ inputs.NODE_VERSION }}
+
+    # Begin Backward compatible setup
```

Comment:
We will bump the action version so this isn't needed and can be removed

Reply:
Good call out will remove.